R/ifelse_censor_linter.R

```diff
@@ -0,0 +1,45 @@
+#' Block usage of ifelse where pmin or pmax is more appropriate
+#'
+#' `ifelse(x > M, M, x)` is the same as `pmin(x, M)`, but harder
+#'   to read and requires several passes over the vector.
+#'
+#' The same goes for other similar ways to censor a vector, e.g.
+#'   `ifelse(x <= M, x, M)` is `pmin(x, M)`,
+#'   `ifelse(x < m, m, x)` is `pmax(x, m)`, and
+#'   `ifelse(x >= m, x, m)` is `pmax(x, m)`.
+#'
+#' @evalRd rd_tags("ifelse_censor_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+ifelse_censor_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$xml_parsed_content) == 0L) {
+      return(list())
+    }
+
+    xml <- source_file$xml_parsed_content
+
+    xpath <- glue::glue("//expr[
+      expr[SYMBOL_FUNCTION_CALL[ {xp_text_in_table(ifelse_funs)} ]]
```

Comment:
my refactoring approach always uses grep so it's the same as any other var in namespace to me 😅

what alternative do you have in mind?

Reply:
Maybe adding a comment to the definition of `ifelse_funs` explaining which linters use it is a good safeguard.

Reply:
that would seem to reverse the pain of maintenance to be up-front instead of eventually (which might never happen)... not sure it's an improvement.

How about a comment like "this is used in several places, e.g. in the glue-built string of `ifelse_censor_linter`"?

Reply:
Yeah, sounds good.